src/rider/main/kotlin/me/fornever/avaloniarider/idea/editor/actions/RunnableAssemblySelectorAction.kt

```diff
@@ -273,7 +273,21 @@ class RunnableAssemblySelectorAction(
                 logger.error("Couldn't find runnable project for path $normalizedPath")
             }
             runnableProject
-        }.sortedBy { it.name }
+        }
+            .filter { !isWebAssemblyProject(it) }
+            .sortedWith(compareBy(
+                // Sort Desktop projects first, then others alphabetically
+                { !it.name.endsWith(".Desktop", ignoreCase = true) },
+                { it.name }
+            ))
         return selectableRunnableProjects
     }
+    
+    private fun isWebAssemblyProject(project: RunnableProject): Boolean {
+        // Filter out WebAssembly/Browser projects which cannot be used for XAML preview
+        // These projects use the Microsoft.NET.Sdk.WebAssembly SDK and won't work with the previewer
+        return project.name.endsWith(".Browser", ignoreCase = true) || 
+               project.name.endsWith(".Wasm", ignoreCase = true) ||
+               project.name.contains("WebAssembly", ignoreCase = true)
```

Comment:
I'd really prefer this to filter the projects by some MSBuild property.

Do you know any property that would help us to identify projects as WASM-related? Or maybe we can rely on the target framework, or something?